### PR TITLE
All Goals Page

### LINKED ISSUE #1005: All Goals Page
Create a page displaying all goals without filters - /goals
Current page living at /goals should be moved to root path

All goals page should have an option to share currently selected filters.

### PR DESCRIPTION
**Pull request by awinogradov**
## PR includes

- [x] Bug Fix
- [x] Refactor
- [x] Feature

## Related issues

Resolve #1005

## QA Instructions, Screenshots, Recordings
  
image

image

### CODE REVIEW

File: src/components/DashboardPage/DashboardPage.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        preset: presetData?.data,
+    });
+
+    const { data, isLoading } = trpc.goal.getUserGoals.useQuery(queryState, {
+        keepPreviousData: true,
+        staleTime: refreshInterval,
+    });
+
+    const goals = data?.goals;
+    const meta = data?.meta;
```
Comment (9teen90nine) [inline src/components/DashboardPage/DashboardPage.tsx:70]: Don't we have "prefer destructurization" rule enabled?

File: src/components/DashboardPage/DashboardPage.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    const userFilters = trpc.filter.getUserFilters.useQuery();
+    const shadowPreset = userFilters.data?.filter((f) => f.params === queryString)[0];
```
Comment (9teen90nine) [inline src/components/DashboardPage/DashboardPage.tsx:72]: .find?

File: src/components/DashboardPage/DashboardPage.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+            [key: string]: {
+                project?: ProjectByIdReturnType | null;
+                goals: NonNullable<GoalByIdReturnType>[];
+            };
+        }>((r, g) => {
+            const k = g.projectId;
+
+            if (k) {
+                if (!r[k]) {
+                    r[k] = {
```
Comment (9teen90nine) [inline src/components/DashboardPage/DashboardPage.tsx:86]: That doesn't look good, you probably should use longer names

File: src/components/GoalsPage/GoalsPage.tsx [hunk 4]
```diff
@@ -182,7 +174,7 @@ export const GoalsPage = ({ user, ssrTime, locale }: ExternalPageProps) => {
     const description =
         currentPreset && currentPreset.description
             ? currentPreset.description
-            : tr('This is your personal goals bundle');
+            : tr('This is all goals across all projects');
```
Comment (9teen90nine) [inline src/components/GoalsPage/GoalsPage.tsx:177]: These are

**Review by 9teen90nine — COMMENTED** [review]
Some minors

File: src/components/DashboardPage/DashboardPage.tsx (hunk 1 above)
Comment (awinogradov) [reply src/components/DashboardPage/DashboardPage.tsx:70]: We have, but data may be null

Comment (9teen90nine) [reply src/components/DashboardPage/DashboardPage.tsx:70]: `const { goals, meta } = data || {}` ?

File: src/components/DashboardPage/DashboardPage.tsx (hunk 2 above)
Comment (awinogradov) [reply src/components/DashboardPage/DashboardPage.tsx:72]: https://github.com/taskany-inc/issues/issues/1094

File: src/components/DashboardPage/DashboardPage.tsx (hunk 3 above)
Comment (awinogradov) [reply src/components/DashboardPage/DashboardPage.tsx:86]: https://github.com/taskany-inc/issues/issues/1095

File: src/components/DashboardPage/DashboardPage.tsx (hunk 1 above)
Comment (awinogradov) [reply src/components/DashboardPage/DashboardPage.tsx:70]: Yep, why not. But this is not new code. It was moved from GoalsPage... Let's do it in https://github.com/taskany-inc/issues/issues/1096

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**